packages/carequality-sdk/src/client/__tests__/carequality-fhir.test.ts

```diff
@@ -0,0 +1,230 @@
+  return new CarequalityManagementApiFhirMock({
+    apiKey,
+    apiMode,
+  });
+}
+
+describe("CarequalityManagementApiFhir", () => {
+  const api = makeCarequalityManagementAPI();
+  if (!api) {
+    console.log("WARNING: Skipping tests because the CQ API is not configured");
+    it.skip("skipping tests", () => {});
+    return;
+  }
+  const managementOid = getEnvVarOrFail("CQ_MANAGEMENT_ORG_OID");
+
+  let oid: string | undefined = undefined;
+  function getOid(): string {
+    if (oid) return oid;
+    if (getApiMode() === APIMode.production) {
+      oid = getEnvVarOrFail("CQ_TEST_ORG_OID");
+      if (!oid) throw new Error("CQ_TEST_ORG_OID is not set");
+    } else {
+      oid = makeOid();
+    }
+    return oid;
+  }
+  function makeLastPart(): number {
+    return faker.number.int({ min: 1, max: 999_999_999 });
+  }
+  function makeOid(): string {
+    return managementOid + "." + orgOidNode + "." + makeLastPart();
+  }
+  function makeUnrelatedOid(): string {
+    const parts = managementOid.split(".");
+    if (parts.length < 2) throw new Error("Management OID is not valid");
+    const lastPart = parseInt(parts[parts.length - 1]!);
+    const partsWithoutLast = parts.slice(0, -1);
+    const newPart = makeLastPart();
+    if (newPart === lastPart) throw new Error("Failed to make unrelated OID");
+    return partsWithoutLast.join(".") + "." + newPart;
+  }
+
+  async function makeOrganization(oid = getOid()): Promise<OrganizationWithId> {
+    const orgFhir = getOrganizationFhirTemplate({
+      oid,
+      name: faker.company.name(),
+      addressLine1: faker.location.streetAddress(),
+      city: faker.location.city(),
+      state: faker.location.state(),
+      postalCode: faker.location.zipCode(),
+      lat: faker.location.latitude().toString(),
+      lon: faker.location.longitude().toString(),
+      contactName: faker.person.fullName(),
+      phone: faker.phone.number(),
+      email: faker.internet.email(),
+      role: "Connection",
+      active: false,
+      parentOrgOid: managementOid,
+    });
+    if (orgFhir.id !== oid) throw new Error("OID mismatch");
+    return {
+      ...orgFhir,
+      id: oid,
+    };
+  }
+
+  let primaryOrg: OrganizationWithId | undefined;
+  let secondaryOrg: OrganizationWithId | undefined;
+  let unrelatedOidOrg: OrganizationWithId | undefined;
+
+  afterAll(async () => {
+    if (secondaryOrg) {
+      try {
+        await api.deleteOrganization(secondaryOrg.id);
+      } catch (error) {
+        console.error(`Failed to delete secondary org (${secondaryOrg.id})`, error);
+      }
+    }
+    if (unrelatedOidOrg) {
+      try {
+        await api.deleteOrganization(unrelatedOidOrg.id);
+      } catch (error) {
+        console.error(`Failed to delete unrelated OID org (${unrelatedOidOrg.id})`, error);
+      }
+    }
+  });
+
+  describe("createOrganization", () => {
+    if (getApiMode() === APIMode.production) {
+      it.skip("createOrganization", () => {});
+      return;
```

Comment:
Think I may have asked this already, but why do we have tests running in production?

Reply:
This helps us spot issues w/ the code that reads from the directory (assuming you're asking specifically about tests for CQ). We can't write to the directory as part of tests against production, but we can read from it - better than nothing. :)